src/components/NewCommentForm.tsx

```
@@ -1,98 +1,228 @@
+  const STORAGE_KEY = `newCommentForm-${postId}`;
+
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+
+  const [errors, setErrors] = useState<{
+    name?: string;
+    email?: string;
+    body?: string;
+  }>({});
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [submitted, setSubmitted] = useState(false);
+
+  useEffect(() => {
+    const saved = localStorage.getItem(STORAGE_KEY);
+
+    if (saved) {
+      const {
+        name: savedName,
+        email: savedEmail,
+        body: savedBody,
+        submitted: savedSubmitted,
+      } = JSON.parse(saved);
+
+      setName(savedName);
+      setEmail(savedEmail);
+      setBody(savedBody);
+      setSubmitted(savedSubmitted);
+    }
+  }, [STORAGE_KEY]);
+
+  useEffect(() => {
+    localStorage.setItem(
+      STORAGE_KEY,
+      JSON.stringify({ name, email, body: '', submitted: false }),
+    );
+  }, [name, email, body, submitted, STORAGE_KEY]);
+
+  const validate = () => {
+    const newErrors: typeof errors = {};
+
+    if (!name.trim()) {
+      newErrors.name = 'Name is required';
+    }
+
+    if (!email.trim()) {
+      newErrors.email = 'Email is required';
+    }
+
+    if (!body.trim()) {
+      newErrors.body = 'Enter some text';
+    }
+
+    return newErrors;
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setSubmitted(true);
+
+    const newErrors = validate();
+
+    if (Object.keys(newErrors).length > 0) {
+      setErrors(newErrors);
+
+      return;
+    }
+
+    setIsSubmitting(true);
+    try {
+      const newComment = await client.post<Comment>('/comments', {
+        postId,
+        name,
+        email,
+        body,
+      });
+
+      onAdd(newComment);
+
+      setBody('');
+      setErrors({});
+      setSubmitted(false);
+      localStorage.removeItem(STORAGE_KEY);
+    } catch {
+    } finally {
+      setIsSubmitting(false);
```

The `catch` block in handleSubmit is empty, so API errors are swallowed silently. At minimum `console.error` the error or set an error state so the UI can display a failure notification and allow retry. Silent failures make debugging and tests harder.